controller/metrics/transportwrapper.go

```diff
@@ -0,0 +1,34 @@
+package metrics
+
+import (
+	"net/http"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+)
+
+type metricsRoundTripper struct {
+	roundTripper  http.RoundTripper
+	app           *v1alpha1.Application
+	metricsServer *MetricsServer
+}
+
+func (mrt *metricsRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	resp, err := mrt.roundTripper.RoundTrip(r)
+	if resp != nil {
+		mrt.metricsServer.IncKubernetesRequest(mrt.app, resp.StatusCode)
```

Comment:
Are we ignoring request failures here?

Reply:
We don't need to handle the error here just need to let it bubble up in stack. If something happened and k8s call failed without status code we still want to increment counter. Changed the code to use status code '0' to indicate that we attempted the k8s request and did not get any response code.